edi_oca/models/edi_exchange_record.py

```diff
@@ -209,7 +211,9 @@ def record(self):
             return None
         if not self.model and self.parent_id:
             return self.parent_id.record
-        return self.env[self.model].browse(self.res_id).exists()
+        return self.env[self.model].search(
+            [("id", "=", self.res_id), ("company_id", "in", self.env.companies.ids)]
+        )
```

Comment:
Proposal to fix method when related model does not have "company_id" field:
```suggestion
        rec = self.env[self.model].browse(self.res_id).exists()
        if rec and "company_id" in rec and rec.sudo().company_id not in self.env.companies.ids:
            # Return empty recordset if company does not match
            rec = rec[:0]
        return rec
```